### DISCUSSION TITLE
How is ruby code colored with this theme in neovim?

### DISCUSSION THREAD

**Discussion by Uaitt** [rmehri01/onenord.nvim#73] [General]
Hey @rmehri01, thank you for this amazing theme!!

I am a Ruby on Rails developer and I mostly use vscode for my day-to-day work. Thus, I have already installed the [vscode port](https://github.com/s1e2b3i4/onenord-vscode) of this theme a few weeks ago and I have played with it quite a bit. I know that that extension is still very young and is probably not 1:1 completely with this neovim theme (it seems like the syntax highlighting could be heavily improved, at least for ruby) , so I thought that I could do some contribution there :) 

The only problem is that I don't know how this theme exactly looks like with ruby code in neovim, and I don't think I want to get through the struggle of setting up a whole neovim ide just for this xD

Thus, I am here to ask if you (or somebody else) could share a few screenshots of ruby code (possibly with LSP and treesitter enabled) so that I can open a contribution on the vscode port and make it match even more this beautiful colorscheme :)

**Comment by rmehri01** [comment]
Hey there!

Yeah the extension is definitely in its early stages, I've seen it work decently for a few languages like Rust and Python but the language support still needs some work.

Thanks for wanting to help out! I'll post examples of both JetBrains and Neovim, let me know if you want any specific code, I'm just using the JetBrains sample since it seems to highlight many of the language features. Feel free to change anything in any of the themes since I'm not a Ruby developer :)

Neovim (treesitter only):

<img width="1582" alt="Screenshot 2023-09-19 at 8 56 17 AM" src="https://github.com/rmehri01/onenord.nvim/assets/52933714/9326c2fc-fb68-4e7c-8282-bfcb929cbb0c">

JetBrains:

<img width="1582" alt="Screenshot 2023-09-19 at 8 50 11 AM" src="https://github.com/rmehri01/onenord.nvim/assets/52933714/8b63570e-f149-431f-8be0-5f849b544230">

I'm also not sure if adding an lsp would make a difference in Neovim, I'll try it out and post it here if so.

Edit: Seems that the lsp (solargraph) didn't make much of a difference, presumably because it doesn't support semantic highlighting, which may also be the reason it looks poorer in VSCode.

**Reply by Uaitt** [reply]
Hey thanks for the screenshots!

Just to let you know, many people in the ruby community have stopped using the solargraph lsp in favour or [Shopify ruby-lsp](https://github.com/Shopify/ruby-lsp) which adds a bit of [semantic highlighting](https://github.com/Shopify/ruby-lsp/blob/main/SEMANTIC_HIGHLIGHTING.md). Can you add it to your neovim config and see how the highlighting behaves?

Anyway I will start working on the vscode port and I will try to fix as much things as I can :) You will probably see my PR on that repo in the upcoming weeks :)

**Reply by rmehri01** [reply]
Ah okay I wasn't aware of that, sorry! I've attached a screenshot:

Neovim (with lsp):

<img width="1582" alt="Screenshot 2023-09-20 at 7 25 06 AM" src="https://github.com/rmehri01/onenord.nvim/assets/52933714/3e341459-ae48-4458-9449-e39338c73d0a">
